bindings/sundials4py/test/sunmatrix/test_sunmatrix_sparse.py

```diff
@@ -0,0 +1,129 @@
+    # Check shape of data
+    dataA = SUNSparseMatrix_Data(A)
+    assert dataA.shape[0] == nnz
+
+
+def test_clone_matrix(sunctx):
+    rows, cols, nnz = 3, 3, 4
+    A = SUNSparseMatrix(rows, cols, nnz, SUN_CSR_MAT, sunctx)
+    B = SUNMatClone(A)
+    assert B is not None
+
+
+def test_zero_matrix(sunctx):
+    rows, cols, nnz = 3, 3, 4
+    A = SUNSparseMatrix(rows, cols, nnz, SUN_CSR_MAT, sunctx)
+    ret = SUNMatZero(A)
+    assert ret == 0
+
+
+def test_copy_matrix(sunctx):
+    rows, cols, nnz = 3, 3, 4
+    A = SUNSparseMatrix(rows, cols, nnz, SUN_CSR_MAT, sunctx)
+    B = SUNSparseMatrix(rows, cols, nnz, SUN_CSR_MAT, sunctx)
+    dataA = SUNSparseMatrix_Data(A)
+    idx_vals = SUNSparseMatrix_IndexValues(A)
+    idx_ptrs = SUNSparseMatrix_IndexPointers(A)
+    # CSR: row 0: col 0 (1.0), row 1: col 1 (1.0), row 2: col 2 (1.0), row 2: col 0 (2.0)
+    dataA[:] = [1.0, 1.0, 1.0, 2.0]
+    idx_vals[:] = [0, 1, 2, 0]
+    idx_ptrs[:] = [0, 1, 2, 4]
+    ret = SUNMatCopy(A, B)
+    assert ret == 0
+    dataB = SUNSparseMatrix_Data(B)
+    idx_valsB = SUNSparseMatrix_IndexValues(B)
+    idx_ptrsB = SUNSparseMatrix_IndexPointers(B)
+    assert np.allclose(dataB, dataA)
+    assert np.allclose(idx_valsB, idx_vals)
+    assert np.allclose(idx_ptrsB, idx_ptrs)
+
+
+def test_scale_add_matrix(sunctx):
+    rows, cols, nnz = 3, 3, 4
+    A = SUNSparseMatrix(rows, cols, nnz, SUN_CSR_MAT, sunctx)
+    B = SUNSparseMatrix(rows, cols, nnz, SUN_CSR_MAT, sunctx)
+    dataA = SUNSparseMatrix_Data(A)
+    dataB = SUNSparseMatrix_Data(B)
+    idx_vals = SUNSparseMatrix_IndexValues(A)
+    idx_ptrs = SUNSparseMatrix_IndexPointers(A)
+    idx_valsB = SUNSparseMatrix_IndexValues(B)
+    idx_ptrsB = SUNSparseMatrix_IndexPointers(B)
+    # CSR: row 0: col 0 (1.0), row 1: col 1 (1.0), row 2: col 2 (1.0), row 2: col 0 (2.0)
+    dataA[:] = [1.0, 1.0, 1.0, 2.0]
+    dataB[:] = [2.0, 2.0, 2.0, 4.0]
+    idx_vals[:] = [0, 1, 2, 0]
+    idx_ptrs[:] = [0, 1, 2, 4]
+    idx_valsB[:] = [0, 1, 2, 0]
+    idx_ptrsB[:] = [0, 1, 2, 4]
+    ret = SUNMatScaleAdd(3.0, A, B)
+    assert ret == 0
+    # 3*A + B = [3+2, 3+2, 3+2, 6+4] = [5, 5, 5, 10]
+    assert np.allclose(dataA, [5.0, 5.0, 5.0, 10.0])
+
+
+def test_scale_add_identity(sunctx):
+    rows, cols, nnz = 3, 3, 4
+    A = SUNSparseMatrix(rows, cols, nnz, SUN_CSR_MAT, sunctx)
+    dataA = SUNSparseMatrix_Data(A)
+    idx_vals = SUNSparseMatrix_IndexValues(A)
+    idx_ptrs = SUNSparseMatrix_IndexPointers(A)
+    # CSR: row 0: col 0 (2.0), row 1: col 1 (2.0), row 2: col 2 (2.0), row 2: col 0 (2.0)
+    dataA[:] = [2.0, 2.0, 2.0, 2.0]
+    idx_vals[:] = [0, 1, 2, 0]
+    idx_ptrs[:] = [0, 1, 2, 4]
+    ret = SUNMatScaleAddI(3.0, A)
+    assert ret == 0
+    # Diagonal elements should be 3*2+1=7, off-diagonal unchanged
+    # So dataA = [7.0, 7.0, 7.0, 2.0] (assuming diagonal at idx_vals[0:3])
+    # But since idx_vals = [0,1,2,0], diagonal is at positions 0,1,2
+    assert np.allclose(dataA[:3], [7.0, 7.0, 7.0])
```

Comment:
Guessing this is AI not quite getting it right 
```suggestion
    # Diagonal elements should be 3*2+1=7, off-diagonal should be 3*2=6
    np.testing.assert_allclose(dataA, [7.0, 7.0, 7.0, 6.0])
```